sonar-scanner-engine/src/main/java/org/sonar/scanner/scan/DefaultInputModuleHierarchy.java

```diff
@@ -111,8 +108,8 @@ public String relativePath(InputModule module) {
       return null;
     }
     DefaultInputModule inputModule = (DefaultInputModule) module;
-    Path parentBaseDir = parent.getBaseDir().toPath();
-    Path moduleBaseDir = inputModule.getBaseDir().toPath();
+    Path parentBaseDir = parent.getBaseDir();
+    Path moduleBaseDir = inputModule.getBaseDir();
 
     return pathResolver.relativePath(parentBaseDir, moduleBaseDir);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Change this instance-reference to a static reference. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2209)